api/rest/swagger.yaml

```diff
@@ -0,0 +1,796 @@
+                $ref: '#/components/schemas/Peer'
+              example: {"id":"QmSJz3VA8cQVeegqZGsTkKQ4uiBXzWBE4r43KPb8woZbtH","addresses":["/p2p-circuit/ipfs/QmSJz3VA8cQVeegqZGsTkKQ4uiBXzWBE4r43KPb8woZbtH","/ip4/127.0.0.1/tcp/9096/ipfs/QmSJz3VA8cQVeegqZGsTkKQ4uiBXzWBE4r43KPb8woZbtH","/ip4/192.168.86.35/tcp/9096/ipfs/QmSJz3VA8cQVeegqZGsTkKQ4uiBXzWBE4r43KPb8woZbtH","/ip4/10.6.0.2/tcp/9096/ipfs/QmSJz3VA8cQVeegqZGsTkKQ4uiBXzWBE4r43KPb8woZbtH","/ip4/192.168.1.42/tcp/9096/ipfs/QmSJz3VA8cQVeegqZGsTkKQ4uiBXzWBE4r43KPb8woZbtH"],"cluster_peers":["QmSJz3VA8cQVeegqZGsTkKQ4uiBXzWBE4r43KPb8woZbtH"],"cluster_peers_addresses":null,"version":"0.10.0","commit":"","rpc_protocol_version":"/ipfscluster/0.10/rpc","error":"","ipfs":{"id":"QmV8LYMZ1KLRzJacFCXrUjYJdHN4XcdTvLmswdkzwVjfse","addresses":["/ip4/127.0.0.1/tcp/4001/ipfs/QmV8LYMZ1KLRzJacFCXrUjYJdHN4XcdTvLmswdkzwVjfse","/ip4/192.168.86.35/tcp/4001/ipfs/QmV8LYMZ1KLRzJacFCXrUjYJdHN4XcdTvLmswdkzwVjfse","/ip4/10.0.3.1/tcp/4001/ipfs/QmV8LYMZ1KLRzJacFCXrUjYJdHN4XcdTvLmswdkzwVjfse","/ip4/10.6.0.2/tcp/4001/ipfs/QmV8LYMZ1KLRzJacFCXrUjYJdHN4XcdTvLmswdkzwVjfse","/ip4/192.168.1.42/tcp/4001/ipfs/QmV8LYMZ1KLRzJacFCXrUjYJdHN4XcdTvLmswdkzwVjfse","/ip6/::1/tcp/4001/ipfs/QmV8LYMZ1KLRzJacFCXrUjYJdHN4XcdTvLmswdkzwVjfse"],"error":""},"peername":"omou"}
+  /version:
+    get:
+      summary: Get Version of Cluster
+      tags:
+        - cluster
+      description: Get the current version of the IPFS Cluster.
+      responses:
+        '200':
+          description: OK
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  Version:
+                    type: string
+              example: {"Version":"0.10.0"}
+              
+  /add:
+    post:
+      summary: Add a File
+      parameters:
+        - $ref: '#/components/parameters/pinNameParam'
+        - $ref: '#/components/parameters/pinReplicationMinParam'
+        - $ref: '#/components/parameters/pinReplicationMaxParam'
+        - $ref: '#/components/parameters/pinShardSize'
+        - $ref: '#/components/parameters/pinMetaParam'
+        - $ref: '#/components/parameters/hash'
+        - $ref: '#/components/parameters/recursive'
+        - $ref: '#/components/parameters/layout'
+        - $ref: '#/components/parameters/rawLeaves'
+        - $ref: '#/components/parameters/hidden'
+        - $ref: '#/components/parameters/wrap'
+        - $ref: '#/components/parameters/shard'
+        - $ref: '#/components/parameters/progress'
+        - $ref: '#/components/parameters/cidVersion'
+        - $ref: '#/components/parameters/streamChannels'
+        - $ref: '#/components/parameters/noCopy'
+        - $ref: '#/components/parameters/chunker'
+      tags:
+        - add
+      description: Add a file to IPFS via IPFS Cluster
+      requestBody:
+        content:
+          multipart/form-data:
+            schema:
+              type: string
+              format: binary
+      responses:
```

Comment:
I don't know if swagger allows to express that this is a streaming response and that, in that case, a trailing header is used for errors. Otherwise this should be noted in the description at least.

Reply:
seems like swagger ui is not capable of handling these headers `Trailer`, `Transfer-Encoding` as they are forbidden by browsers 
https://developer.mozilla.org/en-US/docs/Glossary/Forbidden_header_name 
https://swagger.io/docs/open-source-tools/swagger-ui/usage/limitations/

Reply:
Those are response headers. The docs you link are not related since they are about browsers not allowing some headers to be modified programatically (when making requests).